dotnet_sdk/lib/dependabot/dotnet_sdk/update_checker/latest_version_finder.rb

```diff
@@ -0,0 +1,140 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "excon"
+require "sorbet-runtime"
+
+require "dependabot/dotnet_sdk/requirement"
+require "dependabot/dotnet_sdk/version"
+require "dependabot/registry_client"
+require "dependabot/update_checkers/base"
+
+module Dependabot
+  module DotnetSdk
+    class UpdateChecker < Dependabot::UpdateCheckers::Base
+      class LatestVersionFinder
+        extend T::Sig
+
+        RELEASES_INDEX_URL = "https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json"
+
+        sig { params(dependency: Dependabot::Dependency, ignored_versions: T::Array[String]).void }
+        def initialize(dependency:, ignored_versions:)
+          @dependency = dependency
+          @ignored_versions = ignored_versions
+        end
+
+        sig { returns(T.nilable(Dependabot::Version)) }
+        def latest_version
+          @latest_version ||= T.let(
+            fetch_latest_version,
+            T.nilable(Dependabot::Version)
+          )
+        end
+
+        private
+
+        sig { returns(Dependabot::Dependency) }
+        attr_reader :dependency
+
+        sig { returns(T::Array[String]) }
+        attr_reader :ignored_versions
+
+        sig { returns(T.nilable(Dependabot::Version)) }
+        def fetch_latest_version
+          versions = available_versions
+          versions = filter_prerelease_versions(versions)
+          versions = filter_ignored_versions(versions)
+          versions.max
+        end
```

Comment:
Asking as I don't really know Ruby to grok it from the code, will this support users doing something like this:

1. They use `9.0.100`
2. _A few months pass_
3. Patch Tuesday comes and .NET release `9.0.10x` _and_ `9.0.200`
4. Dependabot updates to `9.0.1xx` because the user doesn't want to ingest new feature bands, and wants to just get runtime updates.

Reply:
Currently, no. I'm fudging it a little and treating it like a semantic version. But is there any reason why someone wouldn't want to always be on the latest feature version?

Reply:
My understanding is that the 1xx feature band is like the "LTSist of LTS" versions, where as new bands can introduce behaviours that can break you, like new analysers.

For example the 8.0.4xx feature band introduced new NuGet Audit functionality, and that can break your build if you use TreatWarningsAsErrors.

I have a vague recollection that 1xx is always what the Linux distros always ship via apt etc., so I imagine that ties into a support choice _somewhere_. Again, Rich Lander probably knows the motivations there.

Reply:
We ship .1xx for the life of a major release. The primary reason is that our source build system (for Linux distros) is locked to .1xx because of compat concerns. This is an agreement we made with Red Hat and Canonical.

Developers that use Visual Studio on Windows and use (for example) containers on Linux often prefer the latest feature band for a different form of compatibility (since VS continues to march the feature bands forward).

Reply:
Okay, it sounds like there could be conflicting requirements depending on the developer's use case or environment. Unfortunately, Dependabot doesn't really have the concept of feature versions.

For simplicity's sake I'll:
- Continue to treat .NET SDK versions as semantic versions
- Add a comment in `dotnet_sdk/version` explaining that we made the conscious choice to do this, and link to [How .NET is versioned][1]

If it's any solace, I implemeted this exact same functionality in Renovate and no-one asked for feature version support 🤷 

[1]: https://learn.microsoft.com/en-us/dotnet/core/versions/#versioning-details